translib/path/path_validator_test.go

```diff
@@ -0,0 +1,155 @@
+	}, {
+		tid:       9, // negative - invalid path
+		path:      "/openconfig-system:system/config/hostname/invalid",
+		errPrefix: "Node invalid not found in the given gnmi path elem",
+	}, { // verify module prefix at list node path
+		tid:     10,
+		path:    "/openconfig-network-instance:network-instances/network-instance[name=*]/protocols/protocol[identifier=OSPF][name=ospfv2]/ospfv2/global/inter-area-propagation-policies/openconfig-ospfv2-ext:inter-area-policy[src-area=0.0.0.1]",
+		resPath: "/openconfig-network-instance:network-instances/network-instance[name=*]/protocols/protocol[identifier=OSPF][name=ospfv2]/ospfv2/global/inter-area-propagation-policies/openconfig-ospfv2-ext:inter-area-policy[src-area=0.0.0.1]",
+	}, { // append module prefix at list node path and verify
+		tid:     11,
+		path:    "/openconfig-network-instance:network-instances/network-instance[name=*]/protocols/protocol[identifier=OSPF][name=ospfv2]/ospfv2/global/inter-area-propagation-policies/inter-area-policy[src-area=0.0.0.1]",
+		resPath: "/openconfig-network-instance:network-instances/network-instance[name=*]/protocols/protocol[identifier=OSPF][name=ospfv2]/ospfv2/global/inter-area-propagation-policies/openconfig-ospfv2-ext:inter-area-policy[src-area=0.0.0.1]",
+	}, {
+		tid:     12,
+		path:    "/openconfig-network-instance:network-instances/network-instance[name=default]/protocols/protocol[identifier=BGP][name=bgp]/bgp/rib/afi-safis/afi-safi[afi-safi-name=L2VPN_EVPN]/openconfig-bgp-evpn-ext:l2vpn-evpn/loc-rib/routes/route[origin=6.6.6.2][path-id=0][prefix=[5\\]:[0\\]:[96\\]:[6601::\\]][route-distinguisher=120.1.1.1:5096]",
+		resPath: "/openconfig-network-instance:network-instances/network-instance[name=default]/protocols/protocol[identifier=BGP][name=bgp]/bgp/rib/afi-safis/afi-safi[afi-safi-name=L2VPN_EVPN]/openconfig-bgp-evpn-ext:l2vpn-evpn/loc-rib/routes/route[origin=6.6.6.2][path-id=0][prefix=[5\\]:[0\\]:[96\\]:[6601::\\]][route-distinguisher=120.1.1.1:5096]",
+	}}
+
+	pathValdtor := NewPathValidator(&(AppendModulePrefix{}), &(AddWildcardKeys{}))
+	for _, tt := range tests {
+		if gPath, err := ygot.StringToPath(tt.path, ygot.StructuredPath); err != nil {
+			t.Error("Error in uri to path conversion: ", err)
+			break
+		} else {
+			pathValdtor.init(gPath)
+			if err := pathValdtor.validatePath(); err != nil {
+				if !strings.HasPrefix(err.Error(), tt.errPrefix) {
+					t.Errorf("Testcase %v failed; error: %v", tt.tid, err)
+				}
+			} else if resPath, err := ygot.PathToString(gPath); resPath != tt.resPath {
+				t.Errorf("Testcase %v failed; error: %v; result path: %v", tt.tid, err, resPath)
+			}
+		}
+	}
+}
+
+func BenchmarkValidatePath1(b *testing.B) {
+	pathValdtor := NewPathValidator(&(AppendModulePrefix{}), &(AddWildcardKeys{}))
+	if gPath, err := ygot.StringToPath("/openconfig-tam:tam/flowgroups/flowgroup[name=*]/config/priority", ygot.StructuredPath); err != nil {
+		b.Error("Error in uri to path conversion: ", err)
+	} else {
```

Comment:
Here and in the below BenchmarkValidatePath functions:
Consider returning in the above if block and removing the else block http://google.github.io/styleguide/go/decisions#indent-error-flow